src/runner.py

```diff
@@ -158,48 +166,56 @@ def get_info(self):
             )
         return runners
 
-    def reconcile(self, num_runners):
+    def reconcile(self, virt_type, quantity):
         """Bring runners in line with target."""
-        active_runners = []
-        stale_runners = []
-        for runner in self.get_info():
-            if runner.is_active:
-                active_runners.append(runner)
-            else:
-                stale_runners.append(runner)
-
-        if stale_runners:
-            runner_names = ", ".join(r.name for r in stale_runners)
-            logger.info(f"Cleaning up stale runners: {runner_names}")
-            for runner in stale_runners:
+        runners = [r for r in self.get_info() if r.virt_type == virt_type]
+        online_runners = [r for r in runners if r.is_online and r.is_local]
+        offline_runners = [r for r in runners if r.is_offline]
+
+        # Clean up stale (offline) runners
+        if offline_runners:
+            runner_names = ", ".join(r.name for r in offline_runners)
+            logger.info(f"Cleaning up stale {virt_type} runners: {runner_names}")
+            for runner in offline_runners:
                 self._remove_runner(runner)
 
-        delta = num_runners - len(active_runners)
+        # Add/Remove runners to match the target quantity
+        delta = quantity - len(online_runners)
         if delta > 0:
-            logger.info(f"Adding {delta} additional runners")
+            logger.info(f"Adding {delta} additional {virt_type} runners")
             for i in range(delta):
-                self.create(image="ubuntu", virt=self.virt_type)
+                self.create(image="ubuntu", virt=virt_type)
         elif delta < 0:
-            active_runners.sort(key=lambda r: r.created_at)
```

Comment:
Does this disagree with the docs?  Which is the right path?  Immediately removing the runners or letting them run eventually and not renewing them when they're gone

> Decreasing the `containers` or `virtual-machines` config values will not shut down runners.

Reply:
Yeah, I should check that the runner is in fact "Idle" and not active before stopping it. Good catch.

Reply:
I have rewritten the readme file - now it should be fully aligned with the actual behavior.